Traceback (most recent call last):
  File "/home/codespace/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/codespace/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/codespace/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/usr/local/python/3.10.13/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/home/codespace/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/home/codespace/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/codespace/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# new row for reordered items 
df['reorder'] = df.item_type.duplicated()
# convert reorder row to boolean value
df['reorder'] = df['reorder'].astype(bool)
#create an ID for each item 
df.rename(columns={'item_id':'item_name'}, inplace=True)
df['item_id'] = pd.factorize(df['item_name'])[0]
#create an ID for each type 
df['type_id'] = pd.factorize(df['item_type'])[0]
#first make a new column for the amount of items per order
df['order_amount'] = df.groupby('order_ID')['amount'].transform('sum')
# make a new column for the price per order
df['order_price'] = df.groupby('order_ID')['price_total'].transform('sum')
------------------

----- stderr -----
/tmp/ipykernel_28579/3658727776.py:7: FutureWarning: factorize with argument that is not not a Series, Index, ExtensionArray, or np.ndarray is deprecated and will raise in a future version.
  df['item_id'] = pd.factorize(df['item_name'])[0]
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
Cell [0;32mIn[3], line 7[0m
[1;32m      5[0m [38;5;66;03m#create an ID for each item [39;00m
[1;32m      6[0m df[38;5;241m.[39mrename(columns[38;5;241m=[39m{[38;5;124m'[39m[38;5;124mitem_id[39m[38;5;124m'[39m:[38;5;124m'[39m[38;5;124mitem_name[39m[38;5;124m'[39m}, inplace[38;5;241m=[39m[38;5;28;01mTrue[39;00m)
[0;32m----> 7[0m df[[38;5;124m'[39m[38;5;124mitem_id[39m[38;5;124m'[39m] [38;5;241m=[39m [43mpd[49m[38;5;241;43m.[39;49m[43mfactorize[49m[43m([49m[43mdf[49m[43m[[49m[38;5;124;43m'[39;49m[38;5;124;43mitem_name[39;49m[38;5;124;43m'[39;49m[43m][49m[43m)[49m[[38;5;241m0[39m]
[1;32m      8[0m [38;5;66;03m#create an ID for each type [39;00m
[1;32m      9[0m df[[38;5;124m'[39m[38;5;124mtype_id[39m[38;5;124m'[39m] [38;5;241m=[39m pd[38;5;241m.[39mfactorize(df[[38;5;124m'[39m[38;5;124mitem_type[39m[38;5;124m'[39m])[[38;5;241m0[39m]

File [0;32m~/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/pandas/core/algorithms.py:765[0m, in [0;36mfactorize[0;34m(values, sort, use_na_sentinel, size_hint)[0m
[1;32m    762[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(values, (ABCIndex, ABCSeries)):
[1;32m    763[0m     [38;5;28;01mreturn[39;00m values[38;5;241m.[39mfactorize(sort[38;5;241m=[39msort, use_na_sentinel[38;5;241m=[39muse_na_sentinel)
[0;32m--> 765[0m values [38;5;241m=[39m [43m_ensure_arraylike[49m[43m([49m[43mvalues[49m[43m,[49m[43m [49m[43mfunc_name[49m[38;5;241;43m=[39;49m[38;5;124;43m"[39;49m[38;5;124;43mfactorize[39;49m[38;5;124;43m"[39;49m[43m)[49m
[1;32m    766[0m original [38;5;241m=[39m values
[1;32m    768[0m [38;5;28;01mif[39;00m (
[1;32m    769[0m     [38;5;28misinstance[39m(values, (ABCDatetimeArray, ABCTimedeltaArray))
[1;32m    770[0m     [38;5;129;01mand[39;00m values[38;5;241m.[39mfreq [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m
[1;32m    771[0m ):
[1;32m    772[0m     [38;5;66;03m# The presence of 'freq' means we can fast-path sorting and know there[39;00m
[1;32m    773[0m     [38;5;66;03m#  aren't NAs[39;00m

File [0;32m~/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/pandas/core/algorithms.py:238[0m, in [0;36m_ensure_arraylike[0;34m(values, func_name)[0m
[1;32m    236[0m     [38;5;28;01mif[39;00m [38;5;28misinstance[39m(values, [38;5;28mtuple[39m):
[1;32m    237[0m         values [38;5;241m=[39m [38;5;28mlist[39m(values)
[0;32m--> 238[0m     values [38;5;241m=[39m [43mconstruct_1d_object_array_from_listlike[49m[43m([49m[43mvalues[49m[43m)[49m
[1;32m    239[0m [38;5;28;01melse[39;00m:
[1;32m    240[0m     values [38;5;241m=[39m np[38;5;241m.[39masarray(values)

File [0;32m~/.cache/pypoetry/virtualenvs/plenty-in-the-pantry-xPFdo8rA-py3.10/lib/python3.10/site-packages/pandas/core/dtypes/cast.py:1598[0m, in [0;36mconstruct_1d_object_array_from_listlike[0;34m(values)[0m
[1;32m   1595[0m [38;5;66;03m# numpy will try to interpret nested lists as further dimensions, hence[39;00m
[1;32m   1596[0m [38;5;66;03m# making a 1D array that contains list-likes is a bit tricky:[39;00m
[1;32m   1597[0m result [38;5;241m=[39m np[38;5;241m.[39mempty([38;5;28mlen[39m(values), dtype[38;5;241m=[39m[38;5;124m"[39m[38;5;124mobject[39m[38;5;124m"[39m)
[0;32m-> 1598[0m [43mresult[49m[43m[[49m[43m:[49m[43m][49m [38;5;241m=[39m values
[1;32m   1599[0m [38;5;28;01mreturn[39;00m result

[0;31mValueError[0m: could not broadcast input array from shape (372,2) into shape (372,)

